### DISCUSSION TITLE
workflows do not work

### DISCUSSION THREAD

**Discussion by newminder** [YetiForceCompany/YetiForceCRM#17079] [General]
I created 4 workflows for one module, all set triggered by [system]. 
1 workflow -- condition: field b = 0  task: update field c= [option1]
2 workflow -- condition: field b greater than 0 and less than field a  task: update field c= [option2]
3 workflow -- condition: field b = field a  task: update field c= [option3]
4 workflow -- condition: field b greater than field a  task: update field c= [option4]

testing results:
change the trigger from system to  Every time a record is saved (eg. creating, modifying), works! all other triggers won't work.

ps: field b is a summary data which collect data and sum from another module (this part works well).

if anyone knows how to solve, please help share your thoughts.

**Comment by mariuszkrzaczkowski** [comment]
system - is called from PHP code, so it won't work if I didn't add proper PHP code

**Reply by newminder** [reply]
can you tell me what codes should be added in the file?
